examples/multiagent/iql.py

```diff
@@ -0,0 +1,283 @@
+    vmas_envs = frames_per_batch // max_steps
+    n_iters = 500  # Number of sampling/training iterations
+    total_frames = frames_per_batch * n_iters
+    memory_size = frames_per_batch
+
+    scenario_name = "balance"
+    env_config = {
+        "n_agents": 3,
+    }
+
+    config = {
+        # RL
+        "gamma": 0.9,
+        "seed": seed,
+        # Sampling,
+        "frames_per_batch": frames_per_batch,
+        "max_steps": max_steps,
+        "vmas_envs": vmas_envs,
+        "n_iters": n_iters,
+        "total_frames": total_frames,
+        "memory_size": memory_size,
+        "vmas_device": vmas_device,
+        # Training
+        "num_epochs": 45,  # optimization steps per batch of data collected
+        "minibatch_size": 4096,  # size of minibatches used in each epoch
+        "lr": 5e-5,
+        "max_grad_norm": 40.0,
+        "training_device": training_device,
+        # Target
+        "tau": 0.005,
+        # Evaluation
+        "evaluation_interval": 20,
+        "evaluation_episodes": 200,
+    }
+
+    model_config = {
+        "shared_parameters": True,
+    }
+
+    # Create env and env_test
+    env = VmasEnv(
+        scenario=scenario_name,
+        num_envs=vmas_envs,
+        continuous_actions=False,
+        max_steps=max_steps,
+        device=vmas_device,
+        seed=seed,
+        # Scenario kwargs
+        **env_config,
+    )
+
+    env_test = VmasEnv(
+        scenario=scenario_name,
+        num_envs=config["evaluation_episodes"],
+        continuous_actions=False,
+        max_steps=max_steps,
+        device=vmas_device,
+        seed=seed,
+        # Scenario kwargs
+        **env_config,
+    )
+    env_config.update({"n_agents": env.n_agents, "scenario_name": scenario_name})
+
+    # Policy
+    net = MultiAgentMLP(
+        n_agent_inputs=env.observation_spec["agents", "observation"].shape[-1],
+        n_agent_outputs=env.action_spec.space.n,
+        n_agents=env.n_agents,
+        centralised=False,
+        share_params=model_config["shared_parameters"],
+        device=training_device,
+        depth=2,
+        num_cells=256,
+        activation_class=nn.Tanh,
+    )
+    module = TensorDictModule(
+        net, in_keys=[("agents", "observation")], out_keys=[("agents", "action_value")]
+    )
+    value_module = QValueModule(
+        action_value_key=("agents", "action_value"),
+        out_keys=[
+            env.action_key,
+            ("agents", "action_value"),
+            ("agents", "chosen_action_value"),
+        ],
+        spec=env.unbatched_action_spec,
+        action_space=None,
+    )
+    qnet = SafeSequential(module, value_module)
+
+    qnet_explore = EGreedyWrapper(
+        qnet,
+        eps_init=0.3,
+        eps_end=0,
+        annealing_num_steps=int(total_frames * (1 / 2)),
+        action_key=env.action_key,
+        spec=env.action_spec,
+    )
+
+    with set_exploration_type(ExplorationType.RANDOM):
+        qnet_explore(env.reset().to(training_device))
```

Comment:
I guess you're init lazy modules, 
perhaps add a comment to tell users what you're doing